### PR TITLE
[root{cp,mv}] Improve documentation of compression settings

### PR DESCRIPTION
**Pull request by maszyman**
Fixes https://github.com/root-project/root/issues/6349

### CODE REVIEW

**Comment by hahnjo** [comment]
@phsft-bot build

File: main/python/rootcp.py [hunk 1]
```diff
@@ -49,10 +55,26 @@ def execute():
 	parser = get_argparse()
 	 # Put arguments in shape
 	sourceList, destFileName, destPathSplit, optDict = cmdLineUtils.getSourceDestListOptDict(parser)
-
+	try:
+		algorithm = optDict["compressalglvl"][0]
+	except (TypeError, IndexError):
+		algorithm = None
+	try:
```
Comment (pcanal) [inline main/python/rootcp.py:62]: This should probably only be run if the previous one succeeded.

File: main/python/rootcp.py [hunk 2]
```diff
@@ -32,13 +32,19 @@
 
 - rootcp -c 1 source.root:hist dest.root
   Change compression factor of 'dest.root' if not existing and copy the histogram named 'hist' from 'source.root' into it.
+
+- rootcp -a 5 9 source.root dest.root
```
Comment (pcanal) [inline main/python/rootcp.py:36]: Why the choice of 'a' followed by 2 arguments?  Did you consider
```
rootcp -c 5 -l 9 source.root dest.root
```
(assuming `-l` is available).

**Comment by vepadulano** [comment]
I re-read the implementation of these functions and I noticed that they call `openROOTFileCompress` which in turn calls into `TFile::SetCompressionSettings` https://github.com/root-project/root/blob/acc96a0132d578519ca1af2ff96dac335e94db0f/main/python/cmdLineUtils.py#L281 

So in practice the already available option `-c` allows to both set the compression algorithm and the compression level via for example `-c 509`. Maybe what we want instead of creating more options is to just better document the existing one? The doc currently says "changing the compression factor" but that is not the full story. We could imagine to point at the documentation of the TFile constructor that explains what the integer passed to the `SetCompressionSettings` function means https://root.cern.ch/doc/master/classTFile.html#ad0377adf2f3d88da1a1f77256a140d60 and also at the doc of https://root.cern.ch/doc/master/structROOT_1_1RCompressionSetting.html which further explains the different compression algorithms available. Maybe as a side-note we could merge/align these two docs so one doesn't need to look for things in two places

**Comment by maszyman** [comment]
Thanks @vepadulano for the comment. I agree with your proposal. I went for an approach with addition of new options because I found `SetCompression{Algorithm,Level}` and assumed that it's desirable to have that level of flexibility. But I think since `SetCompressionSettings` takes care of both, the better user experience is to have single option with better documentation. Hence, I'll revert the changes from b5200e148be517112143fcb5e6d489dbc2d86dd4 and improve documentation.

File: main/python/rootcp.py (hunk 1 above)
Comment (maszyman) [reply main/python/rootcp.py:62]: Agree, but no longer relevant given my comment in https://github.com/root-project/root/pull/14714#issuecomment-1952406831

File: main/python/rootcp.py (hunk 2 above)
Comment (maszyman) [reply main/python/rootcp.py:36]: I wanted to clearly separate settings of algorithm and level. In option `-c` both are convoluted.

Nevertheless, it's no longer relevant given https://github.com/root-project/root/pull/14714#issuecomment-1952406831

File: main/python/rootcp.py [hunk 3]
```diff
@@ -30,9 +30,17 @@
...
-"""
+- rootcp -c 101 source.root:hist dest.root
+  Change compression, if not existing, of 'dest.root' to ZLIB algorithm with compression level 1 and copy the histogram named 'hist' from 'source.root' into it.
+  Meaning of the '-c' argument is given by 'compress = 100 * algorithm + level'.
+  Other examples of usage:
+    * -c 509 : ZSTD with compression level 9
+    * -c 404 : LZ4 with compression level 4
+    * -c 207 : LZMA with compression level 7
+  For more information see https://root.cern.ch/doc/master/classTFile.html#ad0377adf2f3d88da1a1f77256a140d60
+  and https://root.cern.ch/doc/master/structROOT_1_1RCompressionSetting.html.
```
Comment (pcanal) [inline main/python/rootcp.py:41]: This link seem malformed (404 when I click on it).

Comment (maszyman) [reply main/python/rootcp.py:41]: Hmm, I can't really reproduce. Have you tried this from terminal? Perhaps the trailing dot was interpreted as part of the link. I'll remove it.

**Review by vepadulano — APPROVED** [review]
LGTM thank you! Can I ask you to squash the 4 commits into one before we move on with merging?

**Comment by maszyman** [comment]
Done.

Thanks a lot!

File: main/python/rootcp.py (hunk 3 above)
Comment (pcanal) [reply main/python/rootcp.py]: It was the problem.  Thanks.

**Pull request merged**